aidial_sdk/application.py

```diff
@@ -83,13 +92,15 @@ def __init__(
 
         self.add_api_route(
             "/openai/deployments/{deployment_id}/tokenize",
-            self._endpoint_factory("tokenize", TokenizeRequest),
+            self._chat_completion_endpoint_factory("tokenize", TokenizeRequest),
```

Comment:
To be honest, this naming is quite misleading for someone not in context of this repo

Because above there is `self._chat_completion` endpoint, and now there is `chat_completion_endpoint_factory`

And to not being lost at this point, you need to understand, that there is `ChatCompletion` object
And instead of just `chat_completion` method, it also has also some auxiliary features, like `tokenize` and `truncate_prompt`


And the root of problem is that in first place we named class `ChatCompletion`, when its functions are broader that just chat completion

Reply:
This is true, however it's not relevant to this MR and thus should be a subject of another MR.

Renaming classes in SDK is problemantic. We need to be carful to make as less non-backward comptible changes as possible. There are a lot of clients of this library.

Reply:
Yeah, but what's the point of this change `_endpoint_factory` -> `_chat_completion_endpoint_factory`?

I've described, it doesn't add any clarity for outsider

Reply:
And yeah, maybe changing `ChatCompletion` class is not subject of this PR, but it affects it directly:


Basically, we have a problem: 
before we used to implement all logic inside of `ChatCompletion` class, even those logic, that was tied up to chat not strictly 

Now we have new piece of logic - embeddings. Does it ties up to chat less, then `tokenize`? I don't know.

---

So now imagine we want to add new piece of logic

For example, let's imagine that `CriticGPT` got new endpoint, that receive chat, and return some kind of custom response, that shows validation of this response from model

Where does this piece of logic goes?
To `ChatCompletions`? To separate class?

Reply:
> Now we have new piece of logic - embeddings. Does it ties up to chat less, then tokenize? I don't know.

Yes, embeddings has **nothing** to do with chat. This is why it's a separated class.

Reply:
Ok, I agree that moving Embeddings to separate class is valid
But at the same time, I think from now on, we should be super-careful in terms of adding anything to this `ChatCompletion` class

Because tokenizer is good example - it was added, and now it's really hard to pull that off, and not to break backward-compatibility